tests/addr.rs

```diff
@@ -3,75 +3,85 @@ use std::net::{Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4, SocketAddrV6};
-    SocketAddr::V4(SocketAddrV4::new(a, p))
-}
-
-pub fn sa6(a: Ipv6Addr, p: u16) -> SocketAddr {
-    SocketAddr::V6(SocketAddrV6::new(a, p, 0, 0))
-}
-
 #[test]
 fn to_socket_addr_ipaddr_u16() {
     let a = Ipv4Addr::new(77, 88, 21, 11);
     let p = 12345;
     let e = SocketAddr::V4(SocketAddrV4::new(a, p));
-    assert_eq!(Ok(vec![e]), tsa((a, p)));
+    assert_eq!(Ok(vec![e]), blocking_resolve((a, p)));
 }
 
 #[test]
 fn to_socket_addr_str_u16() {
-    let a = sa4(Ipv4Addr::new(77, 88, 21, 11), 24352);
-    assert_eq!(Ok(vec![a]), tsa(("77.88.21.11", 24352)));
+    let a = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(77, 88, 21, 11), 24352));
+    assert_eq!(Ok(vec![a]), blocking_resolve(("77.88.21.11", 24352)));
 
-    let a = sa6(Ipv6Addr::new(0x2a02, 0x6b8, 0, 1, 0, 0, 0, 1), 53);
-    assert_eq!(Ok(vec![a]), tsa(("2a02:6b8:0:1::1", 53)));
+    let a = SocketAddr::V6(SocketAddrV6::new(
+        Ipv6Addr::new(0x2a02, 0x6b8, 0, 1, 0, 0, 0, 1),
+        53,
+        0,
+        0,
+    ));
+    assert_eq!(Ok(vec![a]), blocking_resolve(("2a02:6b8:0:1::1", 53)));
 
-    let a = sa4(Ipv4Addr::new(127, 0, 0, 1), 23924);
+    let a = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 23924));
     #[cfg(not(target_env = "sgx"))]
-    assert!(tsa(("localhost", 23924)).unwrap().contains(&a));
+    assert!(blocking_resolve(("localhost", 23924)).unwrap().contains(&a));
     #[cfg(target_env = "sgx")]
     let _ = a;
 }
 
 #[test]
 fn to_socket_addr_str() {
-    let a = sa4(Ipv4Addr::new(77, 88, 21, 11), 24352);
-    assert_eq!(Ok(vec![a]), tsa("77.88.21.11:24352"));
+    let a = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(77, 88, 21, 11), 24352));
+    assert_eq!(Ok(vec![a]), blocking_resolve("77.88.21.11:24352"));
 
-    let a = sa6(Ipv6Addr::new(0x2a02, 0x6b8, 0, 1, 0, 0, 0, 1), 53);
-    assert_eq!(Ok(vec![a]), tsa("[2a02:6b8:0:1::1]:53"));
+    let a = SocketAddr::V6(SocketAddrV6::new(
+        Ipv6Addr::new(0x2a02, 0x6b8, 0, 1, 0, 0, 0, 1),
+        53,
+        0,
+        0,
+    ));
+    assert_eq!(Ok(vec![a]), blocking_resolve("[2a02:6b8:0:1::1]:53"));
 
-    let a = sa4(Ipv4Addr::new(127, 0, 0, 1), 23924);
+    let a = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 23924));
     #[cfg(not(target_env = "sgx"))]
-    assert!(tsa("localhost:23924").unwrap().contains(&a));
+    assert!(blocking_resolve("localhost:23924").unwrap().contains(&a));
     #[cfg(target_env = "sgx")]
     let _ = a;
 }
 
 #[test]
 fn to_socket_addr_string() {
-    let a = sa4(Ipv4Addr::new(77, 88, 21, 11), 24352);
-    assert_eq!(Ok(vec![a]), tsa(&*format!("{}:{}", "77.88.21.11", "24352")));
-    assert_eq!(Ok(vec![a]), tsa(&format!("{}:{}", "77.88.21.11", "24352")));
-    assert_eq!(Ok(vec![a]), tsa(format!("{}:{}", "77.88.21.11", "24352")));
+    let a = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(77, 88, 21, 11), 24352));
+    assert_eq!(
+        Ok(vec![a]),
+        blocking_resolve(&*format!("{}:{}", "77.88.21.11", "24352"))
```

Comment:
These tests with `format!("{}:{}"` are identical to `blocking_resolve(&format!("{}:{}", "77.88.21.11", "24352"))`. No need to duplicate tests

Reply:
Those code are from std, which test different types: `String`、`&String`、`&str`, I've changed them, new code would be like this:

```Rust
    let s: &str = "77.88.21.11:24352";
    assert_eq!(Ok(vec![a]), blocking_resolve(s));

    let s: &String = &"77.88.21.11:24352".to_string();
    assert_eq!(Ok(vec![a]), blocking_resolve(s));

    let s: String = "77.88.21.11:24352".to_string();
    assert_eq!(Ok(vec![a]), blocking_resolve(s));
```